### PR TITLE
[voq] refine eos fanout function to check fanout link oper status

### PR DESCRIPTION
**Pull request by wenyiz2021**
### Description of PR

This is to fix hidden issue in https://github.com/sonic-net/sonic-mgmt/pull/8468
It could cause failure for test_voq_nbr.py and test_lag_2.py.

eos.py:  current function `check_intf_link_state` is trying to find 'Up' from `show int <interface>` output, but it always returns false because output looks like `Ethernet2/1 is up, line protocol is up (connected)`, probably differs between eos image versions. A more reliable way is to check `show int <interface> | json` and find `connected` in output. Detail is described in code comment.

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

### CODE REVIEW

File: tests/common/devices/eos.py
Comment (slianganet) [inline tests/common/devices/eos.py:160]: I would also caution against adding new code that tries to interpret or directly parse text output from eos commands. The approach in #9127 would be much preferred as the JSON output gives a well-defined model.

Comment (wenyiz2021) [reply tests/common/devices/eos.py]: I agree with you in the json way.
but I think it's necessary to avoid the confusion in the function name as well. if we check for 'connected' then it's checking link oper status only, this is another thing I would want to clarify in this PR

Comment (slianganet) [reply tests/common/devices/eos.py]: Sounds good, I'll let others comment on the preferred naming here. Just wanted to give my 2c on using JSON output.

**Comment by slianganet** [comment]
Thanks for incorporating the feedback and adding comments! I'll let someone with write access give you approval, but otherwise LGTM.

**Review by arlakshm — APPROVED** [review]

**Comment by wenyiz2021** [comment]
thanks @slianganet 

> Thanks for incorporating the feedback and adding comments! I'll let someone with write access give you approval, but otherwise LGTM.

thanks @slianganet and @gechiang for the corporation!

**Pull request merged**
